src/Macros/Recursive.php

```diff
@@ -15,10 +18,14 @@ class Recursive
 {
     public function __invoke()
     {
-        return function (): Collection {
-            return $this->map(function ($value) {
+        return function (float $maxDepth = INF, int $depth = 0): Collection {
```

Comment:
Since in the old code parameters weren't required, and now there are, this is a breaking change.

Reply:
The parameters have a default value. Any existing calls to the recursive method without any parameters continue to work as intended.

Reply:
Right! I must have needed some coffee the moment I review the code previously.